### ISSUE TITLE
Some types of errors are undetected if `-std=c89` is set as a compile flag

### ISSUE THREAD

**Issue by IngoMeyer441** [oclint/oclint#500]
### Issue Summary
Some types of errors are undetected if `-std=c89` is set as a compile flag.

### Environment
- Operation system name and version: macOS High Sierra (10.13.4)
- OCLint version: 0.13.1
- How OCLint is installed: Prebuilt binary from GitHub

### Reproduction Steps
Call oclint with the compilation flag `-std=c89`:
```bash
oclint test.c -- -xc -std=c89
```

#### Sample code

```cpp
int main(void) {
    goto cleanup;
cleanup:
    return 0;
}
```

```bash
oclint test.c -- -xc -std=c89
```

### Expected Behavior
The `goto` statement should be reported as `[basic|P3]`

### Actual Behavior
No errors or warnings are raised.

**Comment by UnrealQuester** [comment]
It looks like our check for the language is not quite enough. In `isLanguageSupported` we check for `ObjC1`, `CPlusPlus` and `C99`. Because clang correctly does not consider c89 to be compatible with c99 we just skip most rules. Sadly `getLangOpts` has no way to check for the other versions, so I have no idea how we would fix this.

If anyone knows how to correctly get the current language version we could just change the check.

**Comment by ryuichi-assistant** [comment]
There hasn't been any activity on this issue recently. Due to the high number of incoming GitHub notifications, we have to clean some of the old issues, as many of them have already been resolved with the latest updates.

Please make sure to update to the latest version and check if that solves the issue. Let us know if that works for you by adding a comment.

**Comment by ryuichi-assistant** [comment]
This issue will be auto-closed because there hasn't been any activity for a few months. Feel free to open a new one if you still experience this problem.
